specs/CookieManagement.md

```diff
@@ -0,0 +1,505 @@
+        /// This only creates a cookie object and it is not added to the cookie
+        /// manager until you call AddOrUpdateCookie.
+        CoreWebView2Cookie CreateCookieWithDetails(String name, String value, String Domain, String Path);
+
+        /// Creates a cookie whose params matches those of the specified cookie.
+        CoreWebView2Cookie CreateCookie(CoreWebView2 cookie);
+
+        /// Gets a list of cookies matching the specific URI.
+        /// If uri is empty string or null, all cookies under the same profile are
+        /// returned.
+        /// You can modify the cookie objects, call
+        /// CoreWebView2CookieManager.AddOrUpdateCookie, and the changes
+        /// will be applied to the webview.
+        Windows.Foundation.IAsyncOperation<IVectorView> GetCookiesAsync(String uri);
+
+        /// Adds or updates a cookie with the given cookie data; may overwrite
+        /// cookies with matching name, domain, and path if they exist.
+        void AddOrUpdateCookie(CoreWebView2Cookie cookie);
+
+        /// Deletes a cookie whose name and domain/path pair
+        /// match those of the specified cookie.
+        void DeleteCookie(CoreWebView2Cookie cookie);
+
+        /// Deletes cookies with matching name and uri.
+        /// Cookie name is required.
+        /// If uri is specified, deletes all cookies with the given name where domain
+        /// and path match provided URI.
+        void DeleteCookies(String name, String uri);
+
+        /// Deletes cookies with matching name and domain/path pair.
+        /// Cookie name is required.
+        /// If domain is specified, deletes only cookies with the exact domain.
+        /// If path is specified, deletes only cookies with the exact path.
+        void DeleteCookiesWithDomainAndPath(String name, String Domain, String Path);
+
+        /// Deletes all cookies under the same profile.
+        /// This could affect other WebViews under the same user profile.
+        void DeleteAllCookies();
+    }
+
+    /// Provides a set of properties that are used to manage a CoreWebView2Cookie.
+    runtimeclass CoreWebView2Cookie
+    {
+        /// Cookie name.
+        String Name { get; };
+
+        /// Cookie value.
+        String Value { get; set; };
+
+        /// The domain for which the cookie is valid.
+        /// The default is the host that this cookie has been received from.
+        /// Note that, for instance, ".bing.com", "bing.com", and "www.bing.com" are
+        /// considered different domains.
+        String Domain { get; };
+
+        /// The path for which the cookie is valid. The default is "/", which means
+        /// this cookie will be sent to all pages on the Domain.
+        String Path { get; };
+
+        /// The expiration date and time for the cookie as the number of seconds since the UNIX epoch.
+        /// The default is -1.0, which means cookies are session cookies by default.

```

Comment:
This comment is slightly wrong for .NET & WinRT. In WinRT Expires should be nullable DateTime (like HttpCookie.Expires) and for .NET use DateTime.MinValue like System.Net.Cookie.